src/Cake.GitHub/Issues/GitHubIssueUpdater.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Cake.Core.Diagnostics;
+using Octokit;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Cake.GitHub
+{
+    internal sealed class GitHubIssueUpdater
+    {
+        private readonly ICakeLog _cakeLog;
+        private readonly IGitHubClient _githubClient;
+
+
+        public GitHubIssueUpdater(ICakeLog cakeLog, IGitHubClient githubClient)
+        {
+            _cakeLog = cakeLog;
+            _githubClient = githubClient;
+        }
+
+
+        public async Task SetMilestoneAsync(string owner, string repository, int number, string milestoneTitle, GitHubSetMilestoneSettings settings)
+        {
+            if (String.IsNullOrWhiteSpace(owner))
+                throw new ArgumentException("Value must not be null or whitespace", nameof(owner));
+
+            if (String.IsNullOrWhiteSpace(repository))
+                throw new ArgumentException("Value must not be null or whitespace", nameof(repository));
+
+            if (number <= 0)
+                throw new ArgumentOutOfRangeException(nameof(number), "Value must not be negative");
+
+            if (String.IsNullOrWhiteSpace(milestoneTitle))
+                throw new ArgumentException("Value must not be null or whitespace", nameof(milestoneTitle));
+
+            _cakeLog.Information($"Setting Milestone for Issue or Pull Request {number}");
```

Comment:
I think it's a good idea to log the milestone here as well

Reply:
Fixed with d947e41be5e65dc7a4304371bb6f690e03f21028